pkg/controller/replication_controller.go

```diff
@@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
 		return nil
 	}
+	if len(controllers) > 1 {
```

Comment:
Why can't we handle all the overlaps the same way, dormant status and best effort rejection at admission time (or let the controllers choose)? The job controller etc will have a different resource and won't come down this watch channel.